packages/calcite-components/src/components/dialog/dialog.e2e.ts

```diff
@@ -802,4 +845,187 @@ describe("calcite-dialog", () => {
 
     expect(await alert.getProperty("embedded")).toBe(true);
   });
+
+  describe("keyboard movement", () => {
+    it("should move properly via arrow keys", async () => {
+      const page = await newE2EPage({
+        html: html`<calcite-dialog width-scale="s" heading="Hello world" drag-enabled open
+          >Hello world!</calcite-dialog
+        >`,
+      });
+      await skipAnimations(page);
+      await page.setViewport({ width: 1200, height: 1200 });
+      await page.waitForChanges();
+      const container = await page.find(`calcite-dialog >>> .${CSS.dialog}`);
+      expect((await container.getComputedStyle()).transform).toBe("matrix(1, 0, 0, 1, 0, 0)");
+
+      await dispatchDialogKeydown({ page, key: "ArrowDown", shiftKey: false });
```

Comment:
Could you use `E2EElement.press` instead of this custom function? I think the main difference would be that you'd need to press shift on the page keyboard before calling `press` on the element (similar to [this](https://github.com/Esri/calcite-design-system/blob/281a86954e266a554dfbbf96c0a641820f11dc56/packages/calcite-components/src/components/color-picker/color-picker.e2e.ts#L2348-L2349)).

Reply:
@jcfranco this isn't working. I think it may be because the function that gets called needs to get boundingClientRect from the element and I'm not sure its getting that when not happening in browser context.

What I tried:

Line 958.
```ts
//await dispatchDialogKeydown({ page, key: "ArrowDown", shiftKey: true });
  await page.keyboard.press("Shift");
  await container.press("ArrowDown");
  await page.waitForChanges();
```

Reply:
Just realized there's an error in the example I linked (will update the test). Could you try this?

```ts
  await page.keyboard.down("Shift"); // down vs press
  await container.press("ArrowDown");
  await page.waitForChanges();
```

Reply:
@jcfranco that still isn't working. Any idea why it wouldn't resize using that method?

```
  ● calcite-dialog › keyboard resize › should resize properly via shift and arrow keys

    expect(received).toBe(expected) // Object.is equality

    Expected: "211px"
    Received: "236px"
```

<img width="976" alt="image" src="https://github.com/user-attachments/assets/be8ae7b0-9f12-46c4-8a5d-c5301b04a090">

Reply:
Not sure. I'll have to explore. Can you create a refactor issue for me? Should be a quick one. 🤞

Reply:
Yeah Ill create one if this PR lands.